_docs/tasks/security/authn-policy.md

```diff
@@ -0,0 +1,288 @@
+
+* Have a Kubernetes cluster with Istio installed (either with or without mTLS, but preferred without as the tasks will show the way to enable mTLS with policy)
+
+* For demo, create two namespaces 'foo' and 'bar', and deploy [httpbin](https://github.com/istio/istio/tree/master/samples/httpbin) and [sleep](https://github.com/istio/istio/tree/master/samples/sleep) with sidecar on both of them. Also, run another sleep without sidecar (to keep it separate, run it in 'legacy' namespace)
+
+```
+kubectl create ns foo
+kubectl apply -f <(istioctl kube-inject --debug -f samples/httpbin/httpbin.yaml) -n foo
+kubectl apply -f <(istioctl kube-inject --debug -f samples/sleep/sleep.yaml) -n foo
+
+kubectl create ns bar
+kubectl apply -f <(istioctl kube-inject --debug -f samples/httpbin/httpbin.yaml) -n bar
+kubectl apply -f <(istioctl kube-inject --debug -f samples/sleep/sleep.yaml) -n bar
+
+kubectl create ns legacy
+kubectl apply -f samples/sleep/sleep.yaml -n legacy
+
+```
+
+* Verifying setup. This probing command send an http request from each of (sleep.foo, sleep.bar, sleep.legacy) to each of (httpbin.foo, httpbin.bar). If Istio was installed without mTLS, all request should return with success code (200) (on the contrary, if Istio was install with mTLS, requests from sleep.legacy should fail).
+
+```
+for from in "foo" "bar" "legacy"; do for to in "foo" "bar"; do kubectl exec $(kubectl get pod -l app=sleep -n ${from} -o jsonpath={.items..metadata.name}) -c sleep -n ${from} -- curl http://httpbin.${to}:8000/ip -s -o /dev/null -w "sleep.${from} to httpbin.${to}: %{http_code}\n"; done; done
+```
+
+```
+sleep.foo to httpbin.foo: 200
+sleep.foo to httpbin.bar: 200
+sleep.bar to httpbin.foo: 200
+sleep.bar to httpbin.bar: 200
+sleep.legacy to httpbin.foo: 200
+sleep.legacy to httpbin.bar: 200
+```
+
+* Also verify that there are no authencation policy in the system
+
+```
+kubectl get policies.authentication.istio.io -n foo
+
+kubectl get policies.authentication.istio.io -n bar
+```
+
+```
+No resources found.
+```
+
+## Enable mTLS for all services in namespace "foo".
+
+Run this command to set namespace-level policy for namespace foo.
+
+```
+cat <<EOF | istioctl create -f -
+apiVersion: "authentication.istio.io/v1alpha1"
+kind: "Policy"
+metadata:
+  name: "enable-mtls"
+  namespace: "foo"
+spec:
+  peers:
+  - mtls:
+EOF  
+```
+
+And verify policy is added:
+```
+kubectl get policies.authentication.istio.io -n foo
+```
+
+```
+NAME          AGE
+enable-mtls   1m
+```
+
+Run the same probing command above. We should see request from sleep.legacy to httpbin.foo start to fail, as the result of mTLS is enabled for httpbin.foo but sleep.legacy doesn't have sidecar to support it. On the other hand, for clients with sidecar (sleep.foo and sleep.bar), Istio automatically config them to using mTLS where requesting http.foo, so they continue to work. Also, requests to "bar" namespace are not affected as policy is for "foo" namespace only.
```

Comment:
request from -> requests from
automatically config them to using where -> automatically configures them to use when

Reply:
Done